src/target/riscv/riscv.h

```diff
@@ -12,12 +12,12 @@ struct riscv_program;
 #include "target/register.h"
 #include "target/semihosting_common.h"
 #include <helper/command.h>
+#include <helper/bits.h>
 
 #define RISCV_COMMON_MAGIC	0x52495356U
 
 /* The register cache is statically allocated. */
-#define RISCV_MAX_HARTS 1024
-#define RISCV_MAX_REGISTERS 5000
+#define RISCV_MAX_HARTS  (int)BIT(20)
```

Comment:
For additional safety when used in expressions, please wrap the whole thing with brackets:

```suggestion
#define RISCV_MAX_HARTS  ((int)BIT(20))
```

Reply:
done